### PR TITLE
Feature/gestpay

### PR DESCRIPTION
**Pull request by sangarbe**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,3 +3,5 @@ bin
 composer.phar
 composer.lock
 phpunit.xml
+.idea/
```
Comment (almacbe) [inline .gitignore]: this has to be part of your global ignore file. http://egorsmirnov.me/2015/05/04/global-gitignore-file.html

File: phpunit.xml.dist [hunk 2]
```diff
@@ -29,8 +28,23 @@
...
     <php>
+        <env name="GESTPAY_SHOP_LOGIN" value="GESPAYXXXX"/>
+        <env name="GESTPAY_API_KEY" value="api-key"/>
         <env name="STRIPE_API_KEY" value="api-key"/>
         <env name="ENABLE_API_INTEGRATION" value="false"/>
+        <env name="SYMFONY_DEPRECATIONS_HELPER" value="weak" />
     </php>
 
+    <listeners>
+        <listener class="Symfony\Bridge\PhpUnit\SymfonyTestsListener">
```
Comment (almacbe) [inline phpunit.xml.dist]: do you really need it?

File: src/PaymentSuite/GestpayBundle/.gitignore [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+# Composer
+composer.phar
+composer.lock
+
+# PHPUnit
+phpunit.xml
+
+# Editors
+.idea/*
```
Comment (almacbe) [inline src/PaymentSuite/GestpayBundle/.gitignore]: same than before

File: src/PaymentSuite/GestpayBundle/phpunit.xml.dist [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+            <exclude>
+                <directory>./Resources/</directory>
+                <directory>./DependencyInjection/</directory>
+                <directory>./vendor/</directory>
+            </exclude>
+        </whitelist>
+    </filter>
+
+    <php>
+        <env name="GESTPAY_SHOP_LOGIN" value="GESPAYXXXX"/>
```
Comment (almacbe) [inline src/PaymentSuite/GestpayBundle/phpunit.xml.dist]: have you added the keys to travis?

File: src/PaymentSuite/RedsysBundle/Controller/PaymentController.php [hunk 5]
```diff
@@ -72,7 +72,7 @@ public function executeAction()
         return new Response(
             $this
                 ->templatingEngine
-                ->render('RedsysBundle:Redsys:process.html.twig', [
+                ->render('GestpayBundle:Redsys:process.html.twig', [
```
Comment (almacbe) [inline src/PaymentSuite/RedsysBundle/Controller/PaymentController.php]: I think the refactor command does something wrong here :-P

File: src/PaymentSuite/RedsysBundle/RedsysMethod.php [hunk 6]
```diff
@@ -18,7 +18,7 @@
 use PaymentSuite\PaymentCoreBundle\PaymentMethodInterface;
 
 /**
- * RedsysMethod class.
+ * GestPayMethod class.
```
Comment (almacbe) [inline src/PaymentSuite/RedsysBundle/RedsysMethod.php]: Same here

File: src/PaymentSuite/RedsysBundle/Services/RedsysManager.php [hunk 7]
```diff
@@ -38,7 +38,7 @@ class RedsysManager
     /**
      * @var RedsysMethodFactory
      *
-     * RedsysMethod factory
+     * GestPayMethod factory
```
Comment (almacbe) [inline src/PaymentSuite/RedsysBundle/Services/RedsysManager.php]: Same here

File: src/PaymentSuite/RedsysBundle/Services/RedsysManager.php [hunk 8]
```diff
@@ -67,7 +67,7 @@ class RedsysManager
      * Construct method for redsys manager.
      *
      * @param RedsysFormTypeBuilder  $redsysFormTypeBuilder  Form Type Builder
-     * @param RedsysMethodFactory    $redsysMethodFactory    RedsysMethod factory
+     * @param RedsysMethodFactory    $redsysMethodFactory    GestPayMethod factory
```
Comment (almacbe) [inline src/PaymentSuite/RedsysBundle/Services/RedsysManager.php]: Same here

File: phpunit.xml.dist (hunk 2 above)
Comment (sangarbe) [reply phpunit.xml.dist]: This is mandatory for the ClockMock to work, i haven't tested it but it's said in the sf docs. However, i'm gonna check it and i'll tell you the results.

File: .gitignore (hunk 1 above)
Comment (sangarbe) [reply .gitignore]: You're right.

File: src/PaymentSuite/RedsysBundle/Controller/PaymentController.php (hunk 5 above)
Comment (sangarbe) [reply src/PaymentSuite/RedsysBundle/Controller/PaymentController.php]: Hmmm yeah it seems so, I was not aware of this. I'll change it and pass the whole test suite

File: phpunit.xml.dist (hunk 2 above)
Comment (sangarbe) [reply phpunit.xml.dist]: Yeah, this is needed. We need to configure de namespaces for the ClockMock, without that it does not work properly.

File: src/PaymentSuite/GestpayBundle/phpunit.xml.dist (hunk 4 above)
Comment (sangarbe) [reply src/PaymentSuite/GestpayBundle/phpunit.xml.dist]: done!

**Pull request merged**